var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -142,12 +153,65 @@ class NetcdfC(AutotoolsPackage):
 
     filter_compiler_wrappers("nc-config", relative_root="bin")
 
+    build_system("cmake", "autotools", default="cmake")
+
+    def setup_run_environment(self, env):
+        if "+zstd" in self.spec:
+            env.append_path("HDF5_PLUGIN_PATH", self.prefix.plugins)
+
+    @property
+    def libs(self):
+        shared = "+shared" in self.spec
+        return find_libraries("libnetcdf", root=self.prefix, shared=shared, recursive=True)
+
+
+class Setup:
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        self.pkg.setup_run_environment(env)
+        # Some packages, e.g. ncview, refuse to build if the compiler path returned by nc-config
+        # differs from the path to the compiler that the package should be built with. Therefore,
+        # we have to shadow nc-config from self.prefix.bin, which references the real compiler,
+        # with a backed up version, which references Spack compiler wrapper.
+        if os.path.exists(self._nc_config_backup_dir):
+            env.prepend_path("PATH", self._nc_config_backup_dir)
+
+
+class BackupStep:
+    @property
+    def _nc_config_backup_dir(self):
+        return join_path(self.pkg.metadata_dir, "spack-nc-config")
+
+    @run_after("install")
+    def backup_nc_config(self):
+        # We expect this to be run before filter_compiler_wrappers:
+        nc_config_file = self.prefix.bin.join("nc-config")
+        if os.path.exists(nc_config_file):
+            mkdirp(self._nc_config_backup_dir)
+            install(nc_config_file, self._nc_config_backup_dir)
+
+
+class CMakeBuilder(CMakeBuilder, BackupStep, Setup):
+    def cmake_args(self):
+        base_cmake_args = [
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define("BUILD_UTILITIES", True),
+            self.define("ENABLE_NETCDF_4", True),
+            self.define_from_variant("ENABLE_DAP", "dap"),
+            self.define("CMAKE_INSTALL_PREFIX", self.prefix),
+            self.define_from_variant("ENABLE_HDF4", "hdf4"),
+            self.define("ENABLE_PARALLEL_TESTS", False),
+        ]
+        if "+parallel-netcdf" in self.spec:
+            base_cmake_args.append(self.define("ENABLE_PNETCDF", True))
+        return base_cmake_args
```

Comment:
There appears to be at least one missing option: 

* `ENABLE_FSYNC`
* `ENABLE_V2_API`


a couple that don't appear to be available in CMake (but are in `configure`):

* `--enable-dynamic-loading`
* `--enable-largefile`

Reply:
Actually I'll just enable other other two for all versions of netcdfc and cmake will ignore those it does not support.